consensus/src/simplex/actors/resolver/ingress.rs

```diff
@@ -1,69 +1,102 @@
 use crate::{
-    simplex::{
-        signing_scheme::Scheme,
-        types::{Notarization, Nullification},
-    },
+    simplex::{signing_scheme::Scheme, types::Voter},
     types::View,
 };
+use bytes::Bytes;
 use commonware_cryptography::Digest;
-use futures::{channel::mpsc, SinkExt};
+use commonware_resolver::{p2p::Producer, Consumer};
+use commonware_utils::sequence::U64;
+use futures::{
+    channel::{mpsc, oneshot},
+    SinkExt,
+};
 use tracing::error;
 
-pub enum Message<S: Scheme, D: Digest> {
-    Fetch {
-        notarizations: Vec<View>,
-        nullifications: Vec<View>,
-    },
-    Notarized {
-        notarization: Notarization<S, D>,
-    },
-    Nullified {
-        nullification: Nullification<S>,
-    },
-    Finalized {
-        // Used to indicate when to prune old notarizations/nullifications.
-        view: View,
-    },
-}
-
 #[derive(Clone)]
 pub struct Mailbox<S: Scheme, D: Digest> {
-    sender: mpsc::Sender<Message<S, D>>,
+    sender: mpsc::Sender<Voter<S, D>>,
 }
 
 impl<S: Scheme, D: Digest> Mailbox<S, D> {
-    pub fn new(sender: mpsc::Sender<Message<S, D>>) -> Self {
+    pub fn new(sender: mpsc::Sender<Voter<S, D>>) -> Self {
         Self { sender }
     }
 
-    pub async fn fetch(&mut self, notarizations: Vec<View>, nullifications: Vec<View>) {
-        if let Err(err) = self
+    pub async fn updated(&mut self, voter: Voter<S, D>) {
+        if let Err(err) = self.sender.send(voter).await {
+            error!(?err, "failed to send voter message");
+        }
+    }
+}
+
+#[derive(Clone)]
+pub struct Handler {
+    sender: mpsc::Sender<Message>,
+}
+
+impl Handler {
+    pub fn new(sender: mpsc::Sender<Message>) -> Self {
+        Self { sender }
+    }
+}
+
+#[derive(Debug)]
+pub enum Message {
+    Deliver {
+        view: View,
+        data: Bytes,
+        response: oneshot::Sender<bool>,
+    },
+    Produce {
+        view: View,
+        response: oneshot::Sender<Bytes>,
+    },
+}
+
+impl Consumer for Handler {
+    type Key = U64;
+    type Value = Bytes;
```

Comment:
This can't be `Voter::` type directly?

Reply:
I tried that briefly but couldn't get it to work for some reason...will try again